store/store.go

```diff
@@ -345,6 +345,101 @@ func (s *Store[H]) HasAt(_ context.Context, height uint64) bool {
 	return height != uint64(0) && s.Height() >= height
 }
 
+// DeleteTo implements [header.Store] interface.
+func (s *Store[H]) DeleteTo(ctx context.Context, to uint64) error {
+	var from uint64
+
+	if tailPtr := s.tailHeader.Load(); tailPtr != nil {
+		from = (*tailPtr).Height()
+	}
+	if from >= to {
+		return nil
+	}
+	if headPtr := s.contiguousHead.Load(); headPtr != nil {
```

Comment:
should we also make inability to load explicit failure? This feels like important sanity check

Reply:
Same as above.